plugins/reporters/spdx/src/main/kotlin/Extensions.kt

```diff
@@ -91,18 +92,47 @@ private fun Package.toSpdxExternalReferences(): List<SpdxExternalReference> {
 }
 
 /**
- * Convert this ORT package to an SPDX package using the information from [licenseInfoResolver]. If [isProject] is
- * `true` then the package is treated as a project.
+ * An enum for the different types of SPDX packages an ORT package can represent.
  */
-internal fun Package.toSpdxPackage(licenseInfoResolver: LicenseInfoResolver, isProject: Boolean = false) =
-    SpdxPackage(
-        spdxId = id.toSpdxId(if (isProject) "Project" else "Package"),
+internal enum class SpdxPackageType(val infix: String, val suffix: String = "") {
+    PROJECT("Project"),
+    BINARY_PACKAGE("Package"),
+    SOURCE_PACKAGE("Package", "source-artifact"),
+    VCS_PACKAGE("Package", "vcs")
+}
+
+/**
+ * Convert this ORT package to an SPDX package. As an ORT package can hold more metadata about its associated artifacts
+ * and origin than an SPDX package, the [type] is used to specify which kind of SPDX package should be created from
+ * respective ORT package metadata. [licenseInfoResolver] is used for license and copyright information, and the
+ * optional [scanResult] and [provenance] are used for [type]-specific information.
```

Comment:
consider stripping the mention of "the optional" as this is obvious from signature and / or doesn't matter for the behavior.

Reply:
Well, the purpose of function docs is to document the function (and its signature), so this "repetition" to "document the obvious" is by design.

Reply:
> so this "repetition" to "document the obvious" is by design.

Sure, by design, not questioning that.